### ISSUE TITLE
Support double clicking to select on word boundaries

### ISSUE THREAD

**Issue by raphCode** [zellij-org/zellij#1258]
I see myself prefer the shift+click selection mode of my terminal emulator, because I can use double-click to select whole words. Double clicking and dragging selects the word under the cursor and extends the selection word-wise.
This reduces the need to aim the cursor at character precision, which is very comfortable.

I need to reach out to zellij's selection only when I want to copy multiple lines, where the pane borders get in the way would I use the terminal's selection.
It would be nice for zellij to support the described double-clicking behaviour as well.

**Comment by vebodev** [comment]
vote +1

**Comment by qiaoboxy** [comment]
vote +1

**Comment by apbyte** [comment]
vote +1

**Comment by thomasaarholt** [comment]
Please just give thumbs up on the top issue to prevent spam :)

Shouldn't this behaviour be entirely controlled by the terminal emulator (app), rather than by zellij?

**Comment by etienne-k** [comment]
> Please just give thumbs up on the top issue to prevent spam :)
> 
> Shouldn't this behaviour be entirely controlled by the terminal emulator (app), rather than by zellij?

While that would be nice, I fear it would no longer be possible to click and drag the selection beyond the currently visible area (i.e. drag up and have Zellij scroll up while selecting text)...

**Comment by barrykp** [comment]
> Please just give thumbs up on the top issue to prevent spam :)
> 
> Shouldn't this behaviour be entirely controlled by the terminal emulator (app), rather than by zellij?

The terminal app does normally provide this functionality (and every terminal I've used has it), but you have to disable zellij's `mouse_mode` to get it back (which I'd rather not do because I like the feature).

**Comment by raphCode** [comment]
The terminal emulator is not aware of zellij's panes and using its selection is unsuited when more than one line should be selected:
![image](https://user-images.githubusercontent.com/15750438/198570394-65c0762e-cf1e-4bb2-81bd-d3ebf5137793.png)

**Comment by emosenkis** [comment]
Triple click to select an entire line is also useful for the same reasons

**Comment by dominz88** [comment]
If this is hard to implement, it would be a good idea to make the fullscreen borderless, this will allow the use of the terminal selection (typically with shift) with double and triple click without the border characters.

**Comment by am-zest** [comment]
Yes, we definitely need Zellij to support double-click to select words, and drag to extend the selection.

As part of that, we would need the ability to define the character set for delimiting 'word' boundaries.

**Comment by avila-m-6** [comment]
> If this is hard to implement, it would be a good idea to make the fullscreen borderless, this will allow the use of the terminal selection (typically with shift) with double and triple click without the border characters.

I find this enough for me in Windows Terminal. `Shift+DblC` for a word selection and `Shift+RgtC` for de-selection.

**Comment by ndsky1003** [comment]
一年了都没解决这问题

**Comment by emosenkis** [comment]
@ndsky1003 This is a volunteer project. No one is entitled to having someone else add any features or even fix any bugs, unless you're [sponsoring](https://github.com/sponsors/imsnif) the author at the $100 tier or more.

**Comment by abrassel** [comment]
hi @imsnif , have there been any updates on this feature? It's my biggest pain point with zellij today.

**Comment by imsnif** [comment]
Hi - first, this is already implemented in main (https://github.com/zellij-org/zellij/pull/3996) and will be released in the next version (no ETA, but I think it should be pretty soon). [My tip jar is here if you're interested](https://github.com/sponsors/imsnif).

Second - friendly suggestion: don't ever mention (tag) maintainers unless you personally know them and have worked with them in the past. Maintainers of popular projects - even ones who do it full-time like myself - get far more notifications than we can handle, on this and many other platforms.

While I'm sure you meant it in an innocent and positive way, this sort of behavior will not win you much friends. You can read more in this excellent oldie-but-goodie blog post: [Entitlement in Open Source](https://mikemcquaid.com/entitlement-in-open-source/)

All the best to you and others eagerly awaiting this much anticipated feature.

**Comment by abrassel** [comment]
Thanks a ton for implementing the feature! I was going to ask if there was any work I could do, but it seems like I was too late.

I am new to open source, and appreciate the feedback :)

**Comment by jovandeginste** [comment]
This works almost perfectly - until there are multi-character unicode symbols in front on the same line (as always, unicode spoils the fun...)

I had a clock symbol (🕑) on my prompt, preceding the current time. I removed this symbol for now, since I value double-click word boundary selection way more than that symbol.

**Comment by imsnif** [comment]
Been working for a while.
